packages/legend-graph/src/graph-manager/GraphManagerState.ts

```diff
@@ -55,10 +55,12 @@ export class BasicGraphManagerState {
   constructor(
     pluginManager: GraphManagerPluginManager,
     logService: LogService,
+    graphManager?: AbstractPureGraphManager,
```

Comment:
Originally I didn't want to add this; however, the V1_PureGraphManager gets created in the `BasicGraphManagerState` constructor (when it calls `buildPureGraphManager`). I wanted to be able to just pass a `V1_EngineServerClient` implementation in the `BasicGraphManagerState` constructor, and then we could just pass that client in the `buildPureGraphManager` function to tell it which client to use when creating the graph manager, but we have a lint rule that states `Do not import file from 'protocol' module in a file from 'graph manager' module`, so we can't use any `V1_` classes in the `GraphManagerState` file. Alternatively, we could pass some boolean flag or enum value in the `BasicGraphManagerState` constructor which will tell `buildPureGraphManager` which client to use, but that also seems a bit messy. Thoughts?

Reply:
I see, because that's the design of graph manager, graph manager relies on some engine, and this engine can be purely local, not tied to any API end point, no network should be made, etc. If we abstract at engine level, this change would not be necessary